account_bank_statement_import_camt_details/models/camt_parser.py

```diff
@@ -41,6 +41,13 @@ def parse_transaction_details(self, ns, node, transaction):
                 './ns:TwnNm', namespaces={'ns': ns})
             if city_node:
                 address_values['city'] = city_node[0].text
+            generic_nodes = address_node[0].xpath(
+                './ns:AdrLine', namespaces={'ns': ns})
+            address_values.update({
+                'address_line_' + str(i): node.text
+                for i, node in enumerate(generic_nodes)
+                if generic_nodes
```

Comment:
If you really mean `if generic_nodes`, why not place that in front of the whole update statement? Now it seems as if you want to check something on each node.